nemo/collections/nlp/data/lm_bert_dataset.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2020 NVIDIA. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+import itertools
+import os
+import pickle
+import random
+
+import h5py
+import numpy as np
+import torch
+from torch.utils.data import DataLoader, DistributedSampler
+
+from nemo import logging
+from nemo.core.classes import Dataset
+from nemo.utils.decorators import experimental
+
+__all__ = ['BertPretrainingPreprocessedDataset', 'BertPretrainingPreprocessedDataloader']
+
+
+def load_h5(input_file):
+    return h5py.File(input_file, "r")
+
+
+# @experimental
+class BertPretrainingPreprocessedDataset(Dataset):
```

Comment:
you should inherit this from nemo.core.Dataset (and put back @experimental until we have tests)